test/expression/eval_test.exs

```diff
@@ -399,4 +399,52 @@ defmodule Expression.EvalTest do
       assert true == Eval.eval!(ast, %{"a" => ~U[2023-11-16 10:40:50.277482Z]})
     end
   end
+
+  describe "handling nil values in date/datetime comparisons" do
+    test "op function directly handles nil in date/datetime comparisons" do
+      # Direct tests of the op function for nil handling
+      date = ~D[2024-01-01]
+      datetime = ~U[2024-01-01 10:40:50.277482Z]
+      date_str = "2024-01-01"
+      datetime_str = "2024-01-01T10:40:50.277482Z"
+
+      # Date with nil string tests
+      assert false == Expression.Eval.op(:>, date, nil)
+      assert false == Expression.Eval.op(:<, date, nil)
+      assert false == Expression.Eval.op(:==, date, nil)
```

Comment:
@davidkhk Can we please also add a case comparing `nil` with `nil`?
```elixir
      assert false == Expression.Eval.op(:==, nil, nil)
      assert false == Expression.Eval.op(:=, nil, nil)
```

Reply:
[Should we turn false when both are `nil`?](https://github.com/turnhub/expression/pull/250/files#r1996195928)